packages/ui/src/redux/slices/settings-dialog.ts

```diff
@@ -2,27 +2,33 @@ import { createSlice, Draft, PayloadAction } from '@reduxjs/toolkit';
 import { SettingsDialogScreen } from 'types';
 
 export interface SettingsDialog {
-  settingsDialogScreen: SettingsDialogScreen;
-  verifiedPassword?: string;
+  screen: SettingsDialogScreen;
+  verifiedPassword?: string | null;
+  loading: boolean;
```

Comment:
1. it's not very clear if we only use one word `loading` here, since we're only using this when changing passing, we should make it more clear by using `onChangingPassword` or something similar.
2. We can still hide the settings modal if clicking outside of the modal (click the backdrop) or hit `ESC` key, we should account for these cases as well.